spec/controllers/settings/registries_controller_spec.rb

```diff
@@ -4,6 +4,13 @@
   let(:user) { create(:user) }
   let(:admin_cert_text) { file_fixture("admin.crt").read.strip }
   let(:pem_cert) { create(:certificate) }
+  let(:pem_cert_text) { pem_cert.certificate.strip }
+  let(:pem_cert_file) do
+    Rack::Test::UploadedFile.new(to_file(pem_cert.certificate), "application/x-x509-user-cert")
```

Comment:
You can use `fixture_file_upload` in controller tests. :wink:

Reply:
Yes, but I create the files on the fly, based on the certificate that is generated by the factory.

So either I have to put those files into the fixture directory (then I can use `fixture_file_upload`) or use this version which just uploads from anywhere.

I could change the `to_file` method to `to_fixture_file` and place the temporary files in the `fixtures` folder instead, which would kind of pollute the folder with a lot of runs though.